From 40a34966bff83f0ba043ea289407ca00e2e935a6 Mon Sep 17 00:00:00 2001
From: "Rafael J. Wysocki" <rjw@sisk.pl>
Date: Tue, 12 Jul 2011 00:39:10 +0200
Subject: PM / Domains: Set device state to "active" during system resume

The runtime PM status of devices in a power domain that is not
powered off in pm_genpd_complete() should be set to "active", because
those devices are operational at this point.  Some of them may not be
in use, though, so make pm_genpd_complete() call pm_runtime_idle()
in addition to pm_runtime_set_active() for each of them.

Signed-off-by: Rafael J. Wysocki <rjw@sisk.pl>
(cherry picked from commit 6f00ff78278fd5d6ac110b6903ee042af2d6af91)

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 drivers/base/power/domain.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/base/power/domain.c b/drivers/base/power/domain.c
index 1f1a7d8..0e7e91b 100644
--- a/drivers/base/power/domain.c
+++ b/drivers/base/power/domain.c
@@ -786,7 +786,9 @@ static void pm_genpd_complete(struct device *dev)
 
 	if (run_complete) {
 		pm_generic_complete(dev);
+		pm_runtime_set_active(dev);
 		pm_runtime_enable(dev);
+		pm_runtime_idle(dev);
 	}
 }
 
-- 
1.7.10.1.362.g242cab3

